app/src/org/commcare/android/tasks/DataPullTask.java

```diff
@@ -133,7 +133,7 @@ public DataPullTask(String username, String password, String server, String keyP
     protected void onCancelled() {
         super.onCancelled();
         if(wasKeyLoggedIn) {
-            CommCareApplication._().logout();
+            CommCareApplication._().getSession().startLogout();

```

Comment:
I've replaced every instance of CommCareApplication._().logout() with the session startLogout.
startLogout eventually triggers finishLogout, which calls the former.
NOTE: finishLogout launches a CommCareHomeActivity intent at the end, prompting the user to log back in. I'm not sure that every instance of CommCareApplication._().logout() solicits this behavior, please check.